flow360/component/simulation/translator/solver_translator.py

```diff
@@ -74,6 +91,70 @@ def rotation_translator(model: Rotation):
     return volume_zone
 
 
+def porous_media_entity_info_serializer(volume):
+    """Porous media entity serializer"""
+    if isinstance(volume, Box):
+        axes = volume.private_attribute_input_cache.axes
+        return {
+            "zoneType": "box",
+            "axes": [list(axes[0]), list(axes[1])],
+            "center": list(volume.center),
+            "lengths": list(volume.size),
+        }
+
+    axes = volume.axes
+    return {
+        "zoneType": "mesh",
+        "zoneName": volume.name,
+        "axes": [list(axes[0]), list(axes[1])],
+    }
+
+
+def porous_media_translator(model: Rotation):
+    """Porous media translator"""
+    porous_medium = remove_units_in_dict(dump_dict(model))
+    porous_medium.pop("volumes")
+    porous_medium.pop("type")
+    porous_medium["DarcyCoefficient"] = list(porous_medium.pop("darcyCoefficient"))
+    porous_medium["ForchheimerCoefficient"] = list(porous_medium.pop("forchheimerCoefficient"))
+    return porous_medium
+
+
+def boundary_translator(model: SurfaceModelTypes, op_acousitc_to_static_pressure_ratio):
+    """Boundary translator"""
+    boundary = remove_units_in_dict(dump_dict(model))
+    boundary.pop("surfaces")
+    if isinstance(model, Wall):
+        boundary["type"] = "WallFunction" if boundary.pop("useWallFunction") else "NoSlipWall"
+        heat_spec = boundary.pop("heat_spec", {"value": None})["value"]
+        if isinstance(model.heat_spec, Temperature):
+            boundary["temperature"] = heat_spec
+        elif isinstance(model.heat_spec, HeatFlux):
+            boundary["heatFlux"] = heat_spec
+    elif isinstance(model, Inflow):
+        boundary["totalTemperatureRatio"] = boundary.pop("totalTemperature")
+        spec = boundary.pop("spec", {"value": None})["value"]
+        if isinstance(model.spec, TotalPressure):
+            boundary["type"] = "SubsonicInflow"
+            boundary["totalPressureRatio"] = spec * op_acousitc_to_static_pressure_ratio
+        elif isinstance(model.spec, MassFlowRate):
+            boundary["type"] = "MassInflow"
+            boundary["massFlowRate"] = spec
+    elif isinstance(model, Outflow):
+        spec = boundary.pop("spec", {"value": None})["value"]
+        if isinstance(model.spec, Pressure):
+            boundary["type"] = "SubsonicOutflowPressure"
+            boundary["staticPressureRatio"] = spec * op_acousitc_to_static_pressure_ratio
+        elif isinstance(model.spec, Mach):
+            pass
+        elif isinstance(model.spec, MassFlowRate):
+            pass
+    elif isinstance(model, Periodic):
+        pass
+
+    return boundary
```

Comment:
I would actually recommend changing how the translation is done at a high level.

`boundary` here initially contains all the V2.0 model fields which basically means that it inherits from the Simulation boundary model. This introduces coupling which we learned a very painful lesson with the legacy model.

Also, by looking at this function, because it is initially already populated, you do not know what keys exactly the returned boundary will have which is perfect place to hide bugs.

Can you return a newly constructed dict instead and enforce this throughout the rest of translation?